File: test/lit/zig/bitwise.zig
// RUN: %cot emit-cir %s | %FileCheck %s

pub fn bitand(a: i32, b: i32) i32 {
    return a & b;
}

pub fn bitor(a: i32, b: i32) i32 {
    return a | b;
}

pub fn bitxor(a: i32, b: i32) i32 {
    return a ^ b;
}

pub fn bitnot(x: i32) i32 {
    return ~x;
}

pub fn shifts(a: i32, b: i32) i32 {
    return (a << b) >> b;
}

// CHECK-LABEL: func.func @bitand
// CHECK: cir.bit_and

// CHECK-LABEL: func.func @bitor
// CHECK: cir.bit_or

// CHECK-LABEL: func.func @bitxor
// CHECK: cir.bit_xor

// CHECK-LABEL: func.func @bitnot
// CHECK: cir.bit_not

// CHECK-LABEL: func.func @shifts
// CHECK: cir.shl
// CHECK: cir.shr

File: test/lit/ts/bitwise.ts
// RUN: %cot emit-cir %s | %FileCheck %s

// Phase 1: bitwise operators

function band(a: number, b: number): number {
    return a & b;
}

function bor(a: number, b: number): number {
    return a | b;
}

function bxor(a: number, b: number): number {
    return a ^ b;
}

function bnot(x: number): number {
    return ~x;
}

// CHECK-LABEL: func.func @band
// CHECK: cir.bit_and

// CHECK-LABEL: func.func @bor
// CHECK: cir.bit_or

// CHECK-LABEL: func.func @bxor
// CHECK: cir.xor

// CHECK-LABEL: func.func @bnot
// CHECK: cir.bit_not
